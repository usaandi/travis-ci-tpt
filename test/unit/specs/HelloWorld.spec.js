import Vue from 'vue';
import HelloWorld from '@/components/HelloWorld';

describe('HelloWorld.vue', () => {
  it('should render correct contents', () => {
    const Constructor = Vue.extend(HelloWorld);
    const vm = new Constructor().$mount();
    expect(vm.$el.querySelector('.hello h1').textContent)
      .toEqual('Welcome to someone homepage!');
  });

  it('we have 4 buttons in div.buttons', () => {
    const Constructor = Vue.extend(HelloWorld);
    const vm = new Constructor().$mount();
    const number = vm.$el.querySelectorAll('div.buttons button').length;
    expect(number).toBe(4);
    expect(number).toMatchSnapshot();
  });

  it('match buttons html', () => {
    const Constructor = Vue.extend(HelloWorld);
    const vm = new Constructor().$mount();
    const html = vm.$el.querySelector('div.buttons').innerHTML;
    console.log('html', html);
    expect(html).toMatchSnapshot();
  });
});
