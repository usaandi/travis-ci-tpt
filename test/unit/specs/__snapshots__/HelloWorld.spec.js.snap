// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HelloWorld.vue match buttons html 1`] = `
<button>b1</button>
<button>b2</button>
<button>b3</button>
<button>b4</button>
`;

exports[`HelloWorld.vue we have 4 buttons in div.buttons 1`] = `4`;
